### DISCUSSION TITLE
Render main calendar in-line with a note with Dataview

### DISCUSSION THREAD

**Discussion by pcause** [davish/obsidian-full-calendar#362] [Feature Requests]
The dataview example you gave gets the items passed to renderCalendar, but I'd like to be able to get access to / display the calendar that I've set up in a "container" so I can embed in a note.  Can you give an example of how to do this and possibly customize the display (start in day/week view, get a specific week, etc)

**Comment by matthewturk** [comment]
I was looking at this as well -- I really want to be able to specify the view for "today's events" in my daily note and connect them to todo's etc. From my reading of the code, it looks like there are some functions in https://github.com/davish/obsidian-full-calendar/blob/main/src/view.ts in `CalendarView` that would be useful to have pulled out so they can be overridden, or used elsewhere, since the main `CalendarView` handles a bunch of the work that `renderCalendar` would need (including getting the local events).  It looks like one might need to override the ability to specify `calendarEl` to get it in place, and maybe pull some of the event handlers out, along with the `calendarSources` (which is exposed in `settings`).

@davish is this something you've been thinking about?  Would a pull request toward this be helpful, or just get in the way?

**Comment by davish** [comment]
This is a great feature request! And it won't be that hard, as @matthewturk mentioned. @matthewturk If you want to take a stab at this, please go ahead!

**Comment by wychwitch** [comment]
> I was looking at this as well -- I really want to be able to specify the view for "today's events" in my daily note and connect them to todo's etc. From my reading of the code, it looks like there are some functions in https://github.com/davish/obsidian-full-calendar/blob/main/src/view.ts in `CalendarView` that would be useful to have pulled out so they can be overridden, or used elsewhere, since the main `CalendarView` handles a bunch of the work that `renderCalendar` would need (including getting the local events). It looks like one might need to override the ability to specify `calendarEl` to get it in place, and maybe pull some of the event handlers out, along with the `calendarSources` (which is exposed in `settings`).
> 
> @davish is this something you've been thinking about? Would a pull request toward this be helpful, or just get in the way?

I came here to suggest the 'todays events' feature as well!! This would be truly fantastic

**Comment by agustinfaure** [comment]
I came across this issue by chance, but this is something I have managed to implement by myself with the plug-in as it is. Here you can see the code that I have in my daily notes template: 

```js
this.container.style.minHeight = "1000px";
const { renderCalendar } = app.plugins.plugins["obsidian-full-calendar"];
let rawData = await dv.pages('"local_calendar"').where(p => p.date = dv.date("{{date:YYYY-MM-DD}}"));
let entries = rawData.map((p) => ({
    allDay: p.allDay,
    start: DateTime.fromISO(p.startTime).toJSDate(),
    end: DateTime.fromISO(p.endTime).toJSDate(),
    title: p.title,
    id: p.path
}));
let isMobile = window.innerWidth < 500;
let calendar = renderCalendar(this.container, [entries.array()]);
calendar.render();
```

See that '"local_calendar"' is the folder where your local calendar should be stored. The pv.pages() pulls a list of all notes in that folder, which are event notes. The .where argument filters by day, and here the baseline template plugin functionality comes into place. Then it is just a matter of parsing the information that's already available in the metadata of the notes as an event for the dataview integration to parse. In this way I see in my daily note my daily schedule without distractions.

I took inspiration from here: https://forum.obsidian.md/t/custom-full-calendar-with-dataview/34133

I hope this is indeed what the discussion was about and I didn't misunderstand the petition. 

Let me know what you think :)

**Comment by pcause** [comment]
thanks and it is great, but i have multiple calendars including ics links (Google/MS365 calendars).  but this is of great help.

**Comment by molliehwhite** [comment]
> > I was looking at this as well -- I really want to be able to specify the view for "today's events" in my daily note and connect them to todo's etc. From my reading of the code, it looks like there are some functions in https://github.com/davish/obsidian-full-calendar/blob/main/src/view.ts in `CalendarView` that would be useful to have pulled out so they can be overridden, or used elsewhere, since the main `CalendarView` handles a bunch of the work that `renderCalendar` would need (including getting the local events). It looks like one might need to override the ability to specify `calendarEl` to get it in place, and maybe pull some of the event handlers out, along with the `calendarSources` (which is exposed in `settings`).
> > @davish is this something you've been thinking about? Would a pull request toward this be helpful, or just get in the way?
> 
> I came here to suggest the 'todays events' feature as well!! This would be truly fantastic

I want to further say this would be great! Adding the functionality of today's events to my daily note would make this perfect!

**Comment by kohane27** [comment]
For anyone interested, there's a [Dataview integration](https://davish.github.io/obsidian-full-calendar/advanced/dataview/) session in the doc. This dataview integration is utterly amazing! I switched from Day Planner to this plugin because I really like the weekly and monthly overview.

However, one big problem for me is that it's `dataviewjs`, meaning that I can't create or edit any events from within other notes. Currently I need to open another pane just for Full Calendar. It'd be amazing if it's a "real and native" embedding of Full Calendar, that is, allowing creating and editing entries from within other pages (like daily note). For example, [Obsidian Tasks](https://github.com/schemar/obsidian-tasks) allows checking and unchecking tasks instead of just displaying a list of tasks.

No hurries however! Two-pane view makes me even more mindful and conscious about how the day goes by. Thank you again for such an amazing plugin!

**Comment by jebeaudoin** [comment]
> It'd be amazing if it's a "real and native" embedding of Full Calendar, that is, allowing creating and editing entries from within other pages (like daily note).

+1 on that. Especially with the introduction of canvas, embedding editable full-calendar views in notes would be amazing.

**Comment by awsharif** [comment]
> See that '"local_calendar"' is the folder where your local calendar should be stored. The pv.pages() pulls a list of all notes in that folder, which are event notes. The .where argument filters by day, and here the baseline template plugin functionality comes into place. Then it is just a matter of parsing the information that's already available in the metadata of the notes as an event for the dataview integration to parse. In this way I see in my daily note my daily schedule without distractions.
> 
> I took inspiration from here: https://forum.obsidian.md/t/custom-full-calendar-with-dataview/34133
> 
> I hope this is indeed what the discussion was about and I didn't misunderstand the petition.
> 
> Let me know what you think :)


Great embed code @agfaur  - for whatever reason this is not working for me... Any ideas where I could be going wrong? I have only changed "local-calendar" to "event-notes" where my events are stored locally...

**Comment by awsharif** [comment]
Hi All,

For anyone else who had the same issue as me, I asked on the Dataview channel on the Obsidian Discord and Krakor gave the below wonderful solution. You should delete the `.where` statement in @agfaur 's solution and swap the `let entries...` to the following

```js
const entries = rawData.map((p) => {
    const startTime = p.startTime.split(':')
    const endTime = p.endTime.split(':')

    const start = dv.luxon.DateTime.fromObject({ ...p.date.c, hour: startTime[0], minute: startTime[1] })
    const end = dv.luxon.DateTime.fromObject({ ...p.date.c, hour: endTime[0], minute: endTime[1] })

    const {title, allDay, completed} = p

    return {
        title,
        allDay,
        completed,
        start: start.toString(),
        end: end.toString(),
    };
});
```

**Comment by agustinfaure** [comment]
I'm sorry for my inactivity, I have not used this in a while... but indeed, my original code was working only partially, I changed it after a while to something very similar to what you're showing, @awsharif. However I stopped embedding the calendar in my daily notes because I ended up finding them too cluttered like that 😄

**Comment by lougreenwood** [comment]
Did anyone make any progress on getting all of the calendars defined in the Full Calendar instance which is configured in Obsidian?

(i.e showing google calendar ics & iCloud calendar events in a dataview)

**Comment by hatmatter** [comment]
So as it stands right now is there any way to render a remote calendar (caldav) into a note? Trying to do what others above are trying and put into my daily note.

**Comment by mathisgauthey** [comment]
I'm looking at this particular feature as well for ma daily-note template. Any update ?
